test/sse.spec.js

```diff
@@ -0,0 +1,228 @@
+            consumer.consume(createStream(streamURL, true));
+            expect(consumer.stream).to.be.an.instanceOf(Promise);
+        });
+
+        it('catches callback errors and propagates them', function() {
+            function willFail() {throw new Error("Cannot connect to stream")}
+            try {
+                consumer.consume(willFail);
+            } catch (err) {
+                expect(err).to.be.an.instanceOf(Error);
+                done();
+            }
+        });
+
+        describe('when there is an existing id', function() {
+            beforeEach(function() {
+                consumer
+                    .consume(createStream(streamURL, true))
+                    .checkpoint({
+                        redisUrl: redisUrl,
+                        messages: 1
+                    });
+                return Promise.delay(50);
+            });
+
+            it('passes the id to the consume function', function(done) {
+                var secondConsumer = new Consumer(redisUrl);
+
+                secondConsumer
+                    .consume(checkId)
+                    .checkpoint({
+                        messages: 1
+                    });
+
+                function checkId(id) {
+                    expect(id).to.exist;
+                    done();
+                    return createStream(streamURL);
+                }
+            });
+
+
+        });
+    });
+
+    describe('#onEvent', function() {
+        describe('when a stream has not been consumed yet', function() {
+            it('throws an error', function() {
+                expect(consumer.onEvent.bind(consumer)).to.throw(Error, /A stream must be consumed first/);
+            });
+        });
+
+        describe('when a stream has been consumed', function() {
+            it('calls the onEvent hook', function(done) {
+                 consumer.consume(createStream(streamURL))
+                     .onEvent(function(data) {
+                         expect(JSON.parse(data.data).foo).to.equal('bar');
+                         expect(data.id).to.equal("0");
+                         done();
+                         return true;
+                     });
+            });
+
+            it('returns the original consumer', function() {
+                var testConsumer = consumer
+                    .consume(createStream(streamURL))
+                    .onEvent(function() {return true;});
+                expect(testConsumer).to.equal(consumer);
+            });
+        });
+    });
+
+    describe('#checkpoint', function() {
+
+        it('creates a checkpoint after receiving 1 message', function() {
+            consumer
+                .consume(createStream(streamThreeURL))
+                .checkpoint({
+                    redisUrl: redisUrl,
+                    messages: 1
+                });
+
+            return Promise.delay(50).then(function(){
+                return redisClient.get('checkpoint')
```

Comment:
perhaps we should think of something else than time to drive off assertions, it's not very deterministic 
how about an optional cb parameter in .checkpoint? 

```js
 consumer
            .consume(createStream(streamThreeURL))
            .checkpoint({
                    redisUrl: redisUrl,
                    messages: 1
                }, 
                function (chunk) {
                     //... perform assertions here
                    done()
                });
```